### Data Analysis with Panda
- Video Game Sales 
- Michael Hendricks
- July 21

In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_csv('data/vgsales.csv')

In [3]:
pub_counts = df[['Publisher']].value_counts()

In [4]:
#pub_counts.head()


In [5]:
pub_counts.head(1)

Publisher      
Electronic Arts    1351
dtype: int64

### 1. The Most common Video Game Publisher is Electronic arts

In [6]:
plat_counts = df[["Platform"]].value_counts()

In [7]:
#plat_counts.head()

In [29]:
most_common_plat = plat_counts.head(1)

In [30]:
most_common_plat

Platform
DS          2163
dtype: int64

### 2. The most common platform is DS

In [9]:
genre_counts = df[['Genre']].value_counts()

In [10]:
#genre_counts.head()

In [11]:
genre_counts.head(1)

Genre 
Action    3316
dtype: int64

### 3. The most common genre is Action

In [12]:
name_global_only = df[['Name','Global_Sales']]

### 4. The top 20 highest grossing games are:

In [13]:
name_global_only.head(20)

Name  Global_Sales
0                                     Wii Sports         82.74
1                              Super Mario Bros.         40.24
2                                 Mario Kart Wii         35.82
3                              Wii Sports Resort         33.00
4                       Pokemon Red/Pokemon Blue         31.37
5                                         Tetris         30.26
6                          New Super Mario Bros.         30.01
7                                       Wii Play         29.02
8                      New Super Mario Bros. Wii         28.62
9                                      Duck Hunt         28.31
10                                    Nintendogs         24.76
11                                 Mario Kart DS         23.42
12                   Pokemon Gold/Pokemon Silver         23.10
13                                       Wii Fit         22.72
14                                  Wii Fit Plus         22.00
15                            Kinect Adventures!         21.82
16                            Grand Theft Auto V         21.40
17                 Grand Theft Auto: San Andreas         20.81
18                             Super Mario World         20.61
19  Brain Age: Train Your Brain in Minutes a Day         20.22

In [14]:
name_NA_only = df[['Name','NA_Sales']]

In [15]:
name_NA_only.median()

NA_Sales    0.08
dtype: float64

In [16]:
df[df['NA_Sales'] == 0.08][['Name','NA_Sales']].sort_values('NA_Sales').tail(10)

Name  NA_Sales
8941                              Galerians      0.08
8939   Disney's Donald Duck: Goin' Quackers      0.08
8936                        Toy Story Racer      0.08
8931                   Kung-Fu: High Impact      0.08
8900                           Cardinal Syn      0.08
8887                         Chessmaster II      0.08
8869            T'ai Fu: Wrath of the Tiger      0.08
8860                 Mirror's Edge Catalyst      0.08
8826                       FIFA Soccer 2002      0.08
11492          Ultimate Shooting Collection      0.08

### 5. For North American video game sales, the median is 0.08 million

In [17]:
df[['NA_Sales']].std()

NA_Sales    0.816683
dtype: float64

### 6. The Highest selling video game of all time is 0.8166 standard deviations above the mean

In [18]:
#df[df['Platform'] == 'Wii'][['Name','Platform','Global_Sales']].sort_values('Global_Sales')

In [19]:
wii_games = df[df['Platform'] == 'Wii'][['Name','Platform','Global_Sales']].sort_values('Global_Sales')

In [20]:
#df[df['Platform'] != 'Wii'][['Name','Platform','Global_Sales']].sort_values('Global_Sales')

In [21]:
non_wii_games = df[df['Platform'] != 'Wii'][['Name','Platform','Global_Sales']].sort_values('Global_Sales')

In [22]:
wii_games.mean()

Global_Sales    0.699404
dtype: float64

In [23]:
non_wii_games.mean()

Global_Sales    0.52339
dtype: float64

### 7. Wii games averages 0.699404 million sold per game compared to only 0.52339 million for other platforms

In [24]:
df[['Year']].min()

Year    1980.0
dtype: float64

In [25]:
df[df['Year'] == 1980.0][['Name','Year']].sort_values('Year')

Name    Year
258         Asteroids  1980.0
544   Missile Command  1980.0
1766          Kaboom!  1980.0
1969         Defender  1980.0
2669           Boxing  1980.0
4025       Ice Hockey  1980.0
5366          Freeway  1980.0
6317           Bridge  1980.0
6896         Checkers  1980.0

### 8. The first 9 video games came out in 1980

In [26]:
df[df['Publisher'] == 'Kemco'][['Name','Year']].sort_values('Year').max()

Name    Woody Woodpecker in Crazy Castle 5
Year                                2005.0
dtype: object

### 9. The Last Kemco game was published in 2005

In [27]:
name_global_genre = df[['Name','Global_Sales','Genre']]

In [28]:
name_global_genre.head(20)

Name  Global_Sales         Genre
0                                     Wii Sports         82.74        Sports
1                              Super Mario Bros.         40.24      Platform
2                                 Mario Kart Wii         35.82        Racing
3                              Wii Sports Resort         33.00        Sports
4                       Pokemon Red/Pokemon Blue         31.37  Role-Playing
5                                         Tetris         30.26        Puzzle
6                          New Super Mario Bros.         30.01      Platform
7                                       Wii Play         29.02          Misc
8                      New Super Mario Bros. Wii         28.62      Platform
9                                      Duck Hunt         28.31       Shooter
10                                    Nintendogs         24.76    Simulation
11                                 Mario Kart DS         23.42        Racing
12                   Pokemon Gold/Pokemon Silver         23.10  Role-Playing
13                                       Wii Fit         22.72        Sports
14                                  Wii Fit Plus         22.00        Sports
15                            Kinect Adventures!         21.82          Misc
16                            Grand Theft Auto V         21.40        Action
17                 Grand Theft Auto: San Andreas         20.81        Action
18                             Super Mario World         20.61      Platform
19  Brain Age: Train Your Brain in Minutes a Day         20.22          Misc

### 10. Tetris is the only 'Puzzle' game in the 7op 20 grossing games